app/Http/Controllers/GeneratePdfController.php

```diff
@@ -0,0 +1,22 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use Illuminate\Http\Request;
+use Illuminate\Support\Facades\Storage;
+
+class GeneratePdfController extends Controller
+{
+    /**
+     * Handle the incoming request.
+     *
+     * @param  \Illuminate\Http\Request  $request
+     * @return \Illuminate\Http\Response
+     */
+    public function __invoke(Request $request)
+    {
+        \Artisan::call('generate');
```

Comment:
Instead of calling the artisan command here, could you extract the code from the generate command to a job that you can dispatch both in this controller and in the command?

Reply:
I'll do that. Why is it better to dispatch a job rather than the artisan command?

Reply:
It's not necessarily better, but it feels cleaner to me. While I have called Artisan from a controller in the past, in my mind, an artisan command via the command line and a controller via a route are both entry points. So, I think calling artisan from a controller feels a bit like calling another controller from a controller. It can be done, but it's not as clean. Extracting a job allows us to have one source for generating the pdf, and you can have multiple entry points (command or controller) call the single source for generating the pdf.